### PR TITLE
(feat) KHP3-3708 Add adapter in form engine to handle creating patient identifiers

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR adds ability to generate patient identifiers. In addition it also adds ability to populate fields with type `patientIdentifer` with existing identifiers.

## Related Issue

https://thepalladiumgroup.atlassian.net/browse/KHP3-3708

### CODE REVIEW

**Review by CynthiaKamau — APPROVED** [review]

File: projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapater.spec.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import { ValidationFactory } from '../form-factory/validation.factory';
+import { ExpressionRunner } from '../expression-runner/expression-runner';
+import { HidersDisablersFactory } from '../form-factory/hiders-disablers.factory';
+import { AlertsFactory } from '../form-factory/show-messages.factory';
+import { JsExpressionHelper } from '../helpers/js-expression-helper';
+import { QuestionFactory } from '../form-factory/question.factory';
+import { ControlRelationsFactory } from '../form-factory/control-relations.factory';
+import { NodeBase } from '../form-factory/form-node';
+import { DebugModeService } from './../services/debug-mode.service';
+import { PatientIdenfierAdapater } from './patient-identifier.adapter';
```
Comment (denniskigen) [inline projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapater.spec.ts:13]: There's a typo in this file name as well as in this import.

File: projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapater.spec.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+const adultForm = require('../../adult.json');
+const mockLocationUuid = "some-location-uuid";
+fdescribe('Patient identifier Value Adapter:', () => {
```
Comment (denniskigen) [inline projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapater.spec.ts:17]: We shouldn't need this `fdescribe` as there's only this one suite in this file.

File: projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapater.spec.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    let adultFormSchema: any;
+    beforeEach(() => {
```
Comment (denniskigen) [inline projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapater.spec.ts:19]: The indentation is off for this file. It should use a tab-width of `2`.

File: projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapter.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import { Injectable } from "@angular/core";
+
+import { NodeBase, GroupNode, ArrayNode } from "../form-factory/form-node";
+import { Form } from "../form-factory";
```
Comment (denniskigen) [inline projects/ngx-formentry/src/form-entry/value-adapters/patient-identifier.adapter.ts:4]: This file ought to use a tab size of `2` as well.

**Pull request merged**